demo-opensuse/owner_alice/create_layout.py

```diff
@@ -0,0 +1,99 @@
+from in_toto.util import import_rsa_key_from_file
+from in_toto.models.layout import Layout
+
+def main():
+  # Load Alice's private key to later sign the layout
+  key_alice = import_rsa_key_from_file("alice")
+  # Fetch and load Bob's and Carl's public keys
+  # to specify that they are authorized to perform certain step in the layout
+  key_bob = import_rsa_key_from_file("../functionary_bob/bob.pub")
+  key_carl = import_rsa_key_from_file("../functionary_carl/carl.pub")
+
+  layout = Layout.read({
+    "_type": "layout",
+    "keys": {
+        key_bob["keyid"]: key_bob,
+        key_carl["keyid"]: key_carl,
+    },
+    "steps": [{
+        "name": "clone",
+        "material_matchrules": [],
+        "product_matchrules": [["CREATE", "connman/_service"],
+            ["CREATE", "connman/connman-1.30.tar.gz"],
+            ["CREATE", "connman/connman-1.30.tar.sign"],
+            ["CREATE", "connman/connman-rpmlintrc"],
+            ["CREATE", "connman/connman.changes"],
+            ["CREATE", "connman/connman.keyring"],
+            ["CREATE", "connman/connman.spec"]],
+        "pubkeys": [key_bob["keyid"]],
+        "expected_command": "git clone https://github.com/shikherverma/connman.git",
+        "threshold": 1,
+      },{
+        "name": "update-changelog",
+        "material_matchrules": [["MATCH", "connman/connman.changes", "WITH", "PRODUCTS", "FROM", "clone"]],
+        # FIXME: CREATE is more like an allow here, is fixed in next version
+        "product_matchrules": [["ALLOW", "connman/connman.changes"]],
+        "pubkeys": [key_bob["keyid"]],
+        "expected_command": "",
+        "threshold": 1,
+      },{
+        "name": "test",
+        "material_matchrules": [],
+        "product_matchrules": [],
+        "pubkeys": [key_carl["keyid"]],
+        "expected_command": "osc build openSUSE_Factory x86_64 connman/connman.spec",
+        "threshold": 1,
+      },{
+        "name": "package",
+        "material_matchrules": [
+            ["MATCH", "connman/_service", "WITH", "PRODUCTS", "FROM", "clone"],
+            ["MATCH", "connman/connman-1.30.tar.gz", "WITH", "PRODUCTS", "FROM", "clone"],
+            ["MATCH", "connman/connman-1.30.tar.sign", "WITH", "PRODUCTS", "FROM", "clone"],
```

Comment:
I'm wondering if we can do *anything* with this detached signature (e.g., check that the hash inside matches the hash reported by connman in this link)